app/models/application_record.rb

```diff
@@ -0,0 +1,3 @@
+class ApplicationRecord < ActiveRecord::Base
```

Comment:
1. Whats this file about?
2. Why need `self.abstract_class = true`?

Reply:
1. All models will inherit from the ApplicationRecord and hence we can define methods that are to be shared across models here.
2. When Rails creates a model as an abstract class it does so without an underlying table. That means that this model can never be instantiated. This is used to solve the issue of having many empty columns for the table when many models inherit from the same parent, each of them having different attributes, each requiring their own column. Since there's no table, it holds no data representation and knows nothing of the data required by its children.